fabric-data-fixer-api-v1/src/main/java/net/fabricmc/fabric/api/datafixer/v1/FabricDataFixes.java

```diff
@@ -0,0 +1,175 @@
+	 */
+	public static void registerFixer(ModContainer mod,
+			@Range(from = 0, to = Integer.MAX_VALUE) int currentVersion,
+			DataFixer dataFixer) {
+		Objects.requireNonNull(mod, "mod cannot be null");
+
+		registerFixer(mod.getMetadata().getId(), currentVersion, dataFixer);
+	}
+
+	/**
+	 * Registers a new data fixer. This method gets the current version from
+	 * the {@code fabric-data-fixer-api-v1:version} field in the {@code custom} object of
+	 * the {@code fabric.mod.json} file of {@code mod}. To specify the version
+	 * manually, use the other overloads.
+	 *
+	 * @param mod            the mod container
+	 * @param dataFixer      the data fixer
+	 * @throws RuntimeException if the version field does not exist or is not a number
+	 * @throws IllegalArgumentException if the data fixer for {@code mod} is already registered
+	 */
+	public static void registerFixer(ModContainer mod, DataFixer dataFixer) {
+		Objects.requireNonNull(mod, "mod cannot be null");
+		registerFixer(mod.getMetadata().getId(), FabricDataFixesInternals.getDataVersionFromMetadata(mod), dataFixer);
+	}
+
+	/**
+	 * Builds and registers a new data fixer.
+	 *
+	 * @param mod              the mod container
+	 * @param builder the data fixer builder
+	 * @throws IllegalArgumentException if the data fixer for {@code mod} is already registered
+	 */
+	public static void buildAndRegisterFixer(ModContainer mod,
+			FabricDataFixerBuilder builder) {
+		Objects.requireNonNull(mod, "mod cannot be null");
+		Objects.requireNonNull(builder, "data fixer builder cannot be null");
+
+		registerFixer(mod.getMetadata().getId(), builder.getDataVersion(),
+				builder.build());
+	}
+
+	/**
+	 * Gets a mod's data fixer.
+	 *
+	 * @param modId the mod ID
+	 * @return the mod's data fixer, or empty optional if the mod hasn't registered one
+	 */
+	public static Optional<DataFixer> getFixer(String modId) {
+		Objects.requireNonNull(modId, "modId cannot be null");
+
+		FabricDataFixesInternals.DataFixerEntry entry = FabricDataFixesInternals.get().getFixerEntry(modId);
+
+		if (entry == null) {
+			return Optional.empty();
+		}
+
+		return Optional.of(entry.dataFixer());
+	}
+
+	/**
+	 * Gets a mod's data version from a {@link NbtCompound}.
+	 *
+	 * @param nbt the NBT compound
+	 * @param modId    the mod ID
+	 * @return the mod's data version, or {@code 0} if the NBT has no data for that mod
+	 */
+	@Contract(pure = true)
+	@Range(from = 0, to = Integer.MAX_VALUE)
+	public static int getModDataVersion(NbtCompound nbt, String modId) {
+		Objects.requireNonNull(nbt, "compound cannot be null");
+		Objects.requireNonNull(modId, "modId cannot be null");
+
+		return FabricDataFixesInternals.getModDataVersion(nbt, modId);
+	}
+
+	/**
+	 * Checks if the data fixer registry is frozen. Data fixers cannot be registered
+	 * after the registry gets frozen.
+	 *
+	 * @return {@code true} if frozen, or {@code false} otherwise.
+	 */
+	@Contract(pure = true)
```

Comment:
Note: Not a pure function since its value is not constant, but that doesn't seem to be exactly what [`Contract.pure` implies](https://github.com/JetBrains/java-annotations/blob/30daf99ad1a0fa0a75edc61a434efdfa2ebc52f5/common/src/main/java/org/jetbrains/annotations/Contract.java#L70-L89)